### PR TITLE
Add GTest framework

### PR DESCRIPTION
**Pull request by Dwordcito**
Setup for Google test framework.

Fixes wazuh/wazuh#5282

### CODE REVIEW

File: src/dbsync/tests/string_helper/main.cpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include "gtest/gtest.h"
+
+int main(int argc, char **argv)
+{
+    ::testing::InitGoogleTest(&argc, argv);
+    int ret = RUN_ALL_TESTS();
```
Comment (konangc87) [inline src/dbsync/tests/string_helper/main.cpp:6]: use {} to initialize a var.

File: src/dbsync/tests/string_helper/string_helper_test.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+void StringHelperTest::SetUp() {};
+
+void StringHelperTest::TearDown() {};
+
+TEST_F(StringHelperTest, CheckReplacement) {
+    std::string string_base("hello_world");
+    auto ret_val { StringHelper::replace_string(string_base, "hello_", "bye_") };
+    EXPECT_EQ(string_base, "bye_world");
+    EXPECT_EQ(ret_val, true);
```
Comment (konangc87) [inline src/dbsync/tests/string_helper/string_helper_test.cpp:12]: EXPECT_TRUE would fit better here.

File: src/dbsync/tests/string_helper/string_helper_test.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+TEST_F(StringHelperTest, CheckNotReplacement) {
+    std::string string_base("hello_world");
+    auto ret_val { StringHelper::replace_string(string_base, "nothing_", "bye_") };
+    EXPECT_EQ(string_base, "hello_world");
+    EXPECT_EQ(ret_val, false);
```
Comment (konangc87) [inline src/dbsync/tests/string_helper/string_helper_test.cpp:19]: EXPECT_FALSE

File: src/dbsync/tests/string_helper/string_helper_test.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#include "string_helper_test.h"
+#include "string_helper.h"
+
+void StringHelperTest::SetUp() {};
+
+void StringHelperTest::TearDown() {};
+
+TEST_F(StringHelperTest, CheckReplacement) {
```
Comment (konangc87) [inline src/dbsync/tests/string_helper/string_helper_test.cpp:8]: { on new line

File: src/dbsync/tests/string_helper/string_helper_test.cpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    std::string string_base("hello_world");
+    auto ret_val { StringHelper::replace_string(string_base, "hello_", "bye_") };
```
Comment (konangc87) [inline src/dbsync/tests/string_helper/string_helper_test.cpp:10]: can be const

File: src/dbsync/tests/string_helper/string_helper_test.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+TEST_F(StringHelperTest, CheckNotReplacement) {
+    std::string string_base("hello_world");
+    auto ret_val { StringHelper::replace_string(string_base, "nothing_", "bye_") };
+    EXPECT_EQ(string_base, "hello_world");
+    EXPECT_EQ(ret_val, false);
+}
+
+TEST_F(StringHelperTest, SplitEmptyString) {
+    auto split_text_vector { StringHelper::split("", '.') };
```
Comment (konangc87) [inline src/dbsync/tests/string_helper/string_helper_test.cpp:23]: can be const

File: src/dbsync/tests/string_helper/string_helper_test.h [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#include "gtest/gtest.h"
+
+class StringHelperTest : public ::testing::Test {
+
+protected:
+
+    StringHelperTest() = default;
+    virtual ~StringHelperTest() = default;
+
+    virtual void SetUp();
```
Comment (tinachoB) [inline src/dbsync/tests/string_helper/string_helper_test.h:10]: Please, change virtual here to use the _override_ specifier in both SetUp and TearDown methods.

File: src/dbsync/tests/string_helper/string_helper_test.h [hunk 8]
```diff
@@ -0,0 +1,12 @@
+#include "gtest/gtest.h"
```
Comment (tinachoB) [inline src/dbsync/tests/string_helper/string_helper_test.h:1]: Add #pragma here or the: 
#ifndef _STRING_HELPER_TEST_H
#define _STRING_HELPER_TEST_H
...
#endif

File: src/dbsync/tests/string_helper/string_helper_test.cpp [hunk 9]
```diff
@@ -0,0 +1,38 @@
+#include "string_helper_test.h"
+#include "string_helper.h"
+
+void StringHelperTest::SetUp() {};
+
+void StringHelperTest::TearDown() {};
+
+TEST_F(StringHelperTest, CheckReplacement) {
+    std::string string_base("hello_world");
```
Comment (tinachoB) [inline src/dbsync/tests/string_helper/string_helper_test.cpp:9]: Please, use {} for initialization

File: src/dbsync/tests/string_helper/string_helper_test.h (hunk 8 above)
Comment (Dwordcito) [reply src/dbsync/tests/string_helper/string_helper_test.h:1]: Good catch!

**Pull request merged**